cmd/manager/main.go

```diff
@@ -259,6 +267,21 @@ func main() {
 		setupLog.Error(err, "unable to set up ready check")
 		os.Exit(1)
 	}
+	setupLog.Info("adding certificate watch", "directory", caCertDir)
+	if err := httputil.AddCertWatch(caCertDir, func() error {
+		ctrl.Log.WithName("cert-pool").Info("reloading certificate pool")
+		certPool, err := httputil.NewCertPool(caCertDir, ctrl.Log.WithName("cert-pool"))
+		if err != nil {
+			return err
+		}
+		ceRecon.CaMutex.Lock()
+		defer ceRecon.CaMutex.Unlock()
+		ceRecon.CaCertPool = certPool
```

Comment:
Instead of updating the cert pool on the ClusterExtensionReconciler, what if we instead used a type that cached this information and that object was passed around?

So instead of having to do something like:
```go
	ceRecon := &controllers.ClusterExtensionReconciler{}

	unpacker := &source.ImageRegistry{
		BaseCachePath: filepath.Join(cachePath, "unpack"),
		// TODO: This needs to be derived per extension via ext.Spec.InstallNamespace
		AuthNamespace: systemNamespace,
		GetCaCertPool: func() (*x509.CertPool, error) {
			ceRecon.CaMutex.RLock()
			defer ceRecon.CaMutex.RUnlock()
			return ceRecon.CaCertPool.Clone(), nil
		},
	}
```

it could instead be:
```go
        certPooler := httputil.NewCertPooler(...)
	unpacker := &source.ImageRegistry{
		BaseCachePath: filepath.Join(cachePath, "unpack"),
		// TODO: This needs to be derived per extension via ext.Spec.InstallNamespace
		AuthNamespace: systemNamespace,
		// Used the existing approach as a way to show the example, but this could
		// probably also just be updated to be a cert pooler instead of a function
		// although keeping it as a function is also fine by me
		GetCaCertPool: func() (*x509.CertPool, error) {
			return certPooler.Get()
		},
	}
```

Reply:
Hmmm.... Would still need to store the pool somewhere, and that place right now is on the reconciler, but doesn't have to be. I can look at this,. but not sure how much "tidier" it would be.

Reply:
The pool would be stored on the `CertPoolWatcher`(?) type